nntrainer/nntrainer_error.h

```diff
@@ -24,13 +24,72 @@
 #define ML_ERROR_RESULT_OUT_OF_RANGE (-ERANGE)
 #endif
 
+#include <functional>
+#include <sstream>
 #include <stdexcept>
+
+#define NNTR_THROW_IF(pred, err) \
+  if (!(pred)) {                 \
+  } else                         \
+    nntrainer::exception::internal::ErrorNotification<err>()
+
+#define NNTR_THROW_IF_CLEANUP(pred, err, cleanup_func) \
+  if (!(pred)) {                                       \
+  } else                                               \
+    nntrainer::exception::internal::ErrorNotification<err>(cleanup_func)
+
 namespace nntrainer {
 
 /// @note underscore_case is used for ::exception to keep in accordance with
 /// std::exception
 namespace exception {
 
+namespace internal {
+
+/**
+ * @brief Error Notification class, error is thrown when the class is destroyed.
+ * DO NOT use this outside as this contains throwing destructor.
+ *
+ * @tparam Err Error type that except cstring as an argument.
```

Comment:
tparam -> param?

Reply:
It's template parameter, so tparam is used 
https://www.doxygen.nl/manual/commands.html#cmdtparam